src/cloud_provider/utilities.rs

```diff
@@ -218,22 +220,47 @@ pub fn generate_supported_version(
 
 // unfortunately some proposed versions are not SemVer like Elasticache (6.x)
 // this is why we need ot have our own structure
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
 pub struct VersionsNumber {
     pub(crate) major: String,
     pub(crate) minor: Option<String>,
     pub(crate) patch: Option<String>,
 }
 
+impl fmt::Display for VersionsNumber {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut version = vec![];
+        version.push(format!("{}", &self.major));
+        if self.minor.is_some() {
+            version.push(format!("{}", &self.minor.clone().unwrap()))
```

Comment:
same here. Why do you use `format!("{}"...)` instead of directly cloning with `self.minor.clone()`.
Plus, you can remove `&self` as it's already a reference. Cf. method signature with `&self` `fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {`